terraform/main.tf

```diff
@@ -1,15 +1,62 @@
-﻿// declarations for the resources to be created 
+// declarations for the resources to be created 
 // this does not include the state S3 bucket as this must be created in order for terraform to init
 
+locals {
+  postgres_config = defaults(var.postgres_config, {
+    allocated_storage           = 100
+    engine                      = "postgres"
+    engine_version              = "14"
+    port                        = 5432
+    backup_retention_period     = 5
+    backup_window               = "04:00-05:00"
+    storage_encrypted           = true
+    publicly_accessible         = true
+    allow_major_version_upgrade = true
+    multi_az                    = false
+    skip_final_snapshot         = true
+  })
+}
 
-module "example" {
-  source = "terraform-aws-modules/s3-bucket/aws"
-  
-  bucket = "gpg-bucket-${terraform.workspace}"
-  acl    = "private"
+resource "aws_security_group" "allow_postgres_connection" {
+  name        = join("_", ["allow_postgres_connection", var.env])
+  description = "Allow Postgres DB traffic"
+  vpc_id      = "vpc-0e9491851e4208470"
+}
 
-  versioning = {
-    enabled = true
-  }
+resource "aws_security_group_rule" "postgres_in" {
+  security_group_id = aws_security_group.allow_postgres_connection.id
+  type              = "ingress"
+  from_port         = local.postgres_config.port
+  to_port           = local.postgres_config.port
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
 
+resource "aws_security_group_rule" "postgres_out" {
+  security_group_id = aws_security_group.allow_postgres_connection.id
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_db_instance" "gpg-dev-db" {
+  allocated_storage           = local.postgres_config.allocated_storage
+  engine                      = local.postgres_config.engine
+  engine_version              = local.postgres_config.engine_version

```

Comment:
Does it matter that this is a newer version than used in the codebase?

Reply:
It should make no difference. Locally I installed the latest version by default and it works without issue, so this should too.